code/modules/unit_tests/food_reagents_check.dm

```diff
@@ -0,0 +1,24 @@
+/// Makes sure that spawned food has reagents and the edible component (or else it can't be eaten).
+/datum/unit_test/food_edibility_check
+
+/datum/unit_test/food_edibility_check/Run()
+	var/list/not_food = list(
+	/obj/item/food/grown,
+	/obj/item/food/grown/mushroom,
+	/obj/item/food/deepfryholder,
+	/obj/item/food/clothing,
+	/obj/item/food/meat/slab/human/mutant,
+	/obj/item/food/grown/shell)
+
+	var/list/food_paths = subtypesof(/obj/item/food) - not_food
+
+	for(var/obj/item/food/food_path as anything in food_paths)
+		var/obj/item/food/spawned_food = new food_path
```

Comment:
```suggestion
		var/obj/item/food/spawned_food = allocate(food_path)
```

`allocate` is something in tests that will automatically delete at the end. I know you're already using `qdel`, but `allocate` provides some good safety (what if GetComponent panics, for some reason).